common/network-common/src/main/java/org/apache/spark/network/sasl/SaslClientBootstrap.java

```diff
@@ -88,8 +91,14 @@ public void doBootstrap(TransportClient client, Channel channel) {
           throw new RuntimeException(
             new SaslException("Encryption requests by negotiated non-encrypted connection."));
         }
+
+        if (conf.saslEncryptionAesEnabled()) {
+          negotiateAes(client, saslClient);
```

Comment:
I think it should be possible. Thanks for the idea!
I am looking at it and may push another update later.